openrtx/src/rtx/OpMode_M17.cpp

```diff
@@ -335,6 +349,28 @@ void OpMode_M17::txState(rtxStatus_t *const status)
         modulator.sendFrame(m17Frame);
     }
 
+    gps_t gps_data;
+    gps_data = state.gps_data;
+    if(encoder.canUpdateLsf() && state.settings.gps_enabled && (gpsTimer == -1 || gpsTimer >= MINIMUM_GPS_TIME_TICKS) && gps_data.fix_type > FIX_TYPE_NOT_AVAIL)
```

Comment:
Two things:

1. For the first initialization, I'd just put 
```
if(state.settings.gps_enabled)
    lsf.setGnssData(&gps_data, M17_GNSS_STATION_HANDHELD);
```
in the block preparing the LSF for the first transmission. This saves you from having to set `gpsTimer` to -1 and have a cumbersome condition checking.

2. Move the `gps_data.fix_type > FIX_TYPE_NOT_AVAIL` check inside the `setGnssData` function: there is the place where you're composing the data block and setting the various validity flags